### PR TITLE
Testrunners tests

### PR DESCRIPTION
**Pull request by Gochad**
Simple tests for `YoutubeTestRunner` object which purpose is running already generated tests through `youtube_generator.py` from generators module.

### CODE REVIEW

File: apptests/test_testrunners.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            "", "audio", "./output", iterations=1, tester=Mock()
+        )
+        return super().setUp()
+
+    # it didnt work - TypeError is raised when value is get from dict
+    def test_none_dict(self):
+        self.assertRaises(TypeError, self.testrunner.save_results(None))
+
+        json_files = [file for file in listdir("apptests") if file.endswith(".json")]
+        self.assertEqual(len(json_files), 2)
```
Comment (Kowalski1024) [inline apptests/test_testrunners.py]: We expect the type `dict[str, Any]`, so I don't think this test is necessary

File: apptests/test_testrunners.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    def test_empty_dict(self):
+        self.assertRaises(KeyError, self.testrunner.save_results({}))
+
+        json_files = [file for file in listdir("apptests") if file.endswith(".json")]
+        self.assertEqual(len(json_files), 2)
+
+    def test_valid_dict(self):
+        path = "simpleTest.json"
+        runner = YouTubeTestRunner(path, "audio", "", iterations=1, tester=Mock())
+        generated_testplan = generate(search_request.get("args"))
```
Comment (Kowalski1024) [inline apptests/test_testrunners.py]: Why don't we use the added file  `simpleTest.json` for this?

File: apptests/Travel&Events_en_CAUQAQ_20231126-201522.json
Comment (Kowalski1024) [inline apptests/Travel&Events_en_CAUQAQ_20231126-201522.json:1]: Why do we have this file if we have `simpleTest.json`?

File: apptests/simpleTest.json
Comment (Kowalski1024) [inline apptests/simpleTest.json:1]: Let's move it to `apptests/data/`

File: apptests/test_testrunners.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        "channelId": "UCkeTA80xeSa9POCGRpWTIfQ",
+        "channelTitle": "travelingisrael.com",
+        "defaultAudioLanguage": "en",
+        "duration": "PT16M18S",
+        "manuallyCreatedTranscripts": ["en"],
+        "publishTime": "2023-10-18T12:38:57Z",
+        "title": "Why are there so many Palestinian casualties? (The Israeli perspective) sub: DE, ES, FR, IT",
+        "videoId": "OF95GenB1JI",
+    },
+]
```
Comment (Kowalski1024) [inline apptests/test_testrunners.py]: Variables are not used or can be replaced by loading the `simpleTest.json` file.

File: apptests/test_testrunners.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+        with patch("builtins.open"), patch(
+            "src.dataclasses.youtube_video.YouTubeVideo.from_dict"
+        ), patch("youtube_runner.YouTubeTestRunner.save_results"), patch(
+            "generators.youtube_generator.generate"
+        ):
+            with self.assertRaises(ValueError) as context:
+                self.runner.run()
+
+        self.assertEqual(str(context.exception), "Transcriber is None")
```
Comment (Kowalski1024) [inline apptests/test_testrunners.py]: I think some of the mock logic can be moved to the setUp method. setUp is called for each test method, so mocks can be created in this method and later slightly modified in the test method.

File: apptests/test_testrunners.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+from src.dataclasses.youtube_video import YouTubeVideo
+from src.differs import jiwer_differ
+from youtube_runner import YouTubeTestRunner
+
+
+class Test_YoutubeTestRunner_run(unittest.TestCase):
+    def setUp(self) -> None:
+        self.tester = Mock()
+        self.runner = YouTubeTestRunner(
+            "Travel&Events_en_CAUQAQ_20231126-201522.json",
```
Comment (Kowalski1024) [inline apptests/test_testrunners.py]: it can be changed to simpleTest.json

File: apptests/test_testrunners.py (hunk 4 above)
Comment (Gochad) [reply apptests/test_testrunners.py]: i moved logic to setup

**Pull request merged**